docs/root/configuration/upstream/health_checkers/thrift.rst

```diff
@@ -0,0 +1,29 @@
+.. _config_health_checkers_thrift:
+
+Thrift Health Checker
+=====================
+
+The Thrift Health Checker (with :code:`envoy.health_checkers.thrift` as name) uses Thrift requests,
+responses and exceptions to check upstream hosts. It sends a request with
+:ref:`method_name <envoy_v3_api_field_extensions.health_checkers.thrift.v3.Thrift.method_name>`
+and expects a success response in exchange. The upstream host can also respond with an exception to cause the check to fail.
+The :ref:`transport <envoy_v3_api_field_extensions.health_checkers.thrift.v3.Thrift.transport>` and
+:ref:`protocol <envoy_v3_api_field_extensions.health_checkers.thrift.v3.Thrift.protocol>` types to be set for each health
+check request must be configured to enable Thrift health checks. The sequence id is always 0 for each health check request.
```

Comment:
> The sequence id is always 0 for each health check request.

Is this true when reusing connections? I also don't see anything in the docs about using a new connection on every check vs reusing the connection (but maybe this is configured at an upper level)....

Reply:
> Is this true when reusing connections? 

Yes, we force the seq id to 0 and not increasing it. btw default is *reusing* connection.
```
void ThriftHealthChecker::ThriftActiveHealthCheckSession::onInterval() {
  ENVOY_LOG(trace, "ThriftActiveHealthCheckSession onInterval");
  if (!client_) {
    ENVOY_LOG(trace, "ThriftActiveHealthCheckSession construct client");
    client_ = parent_.client_factory_.create(
        *this, parent_.transport_, parent_.protocol_, parent_.method_name_, host_,
        /* health checker seq id */ 0, /* fixed_seq_id */ true);
    client_->start();
    expect_close_ = false;
  }

  client_->sendRequest();
}
```

> I also don't see anything in the docs about using a new connection on every check vs reusing the connection (but maybe this is configured at an upper level)....

Right, It's in the upper level. 
https://github.com/envoyproxy/envoy/blob/0b10b055a6e1dbdd600f9a80f8b8d803bdd8ec9c/api/envoy/config/core/v3/health_check.proto#L304

And we close the connection for each check response if we don't want to reuse
```
void ThriftHealthChecker::ThriftActiveHealthCheckSession::onResponseResult(bool is_success) {
  if (is_success) {
    handleSuccess();
  } else {
    // TODO(kuochunghsu): We might want to define retriable response.
    handleFailure(envoy::data::core::v3::ACTIVE, /* retriable */ false);
  }

  if (client_ && !parent_.reuse_connection_) {
    expect_close_ = true;
    client_->close();
  }
}
```

See also `ThriftHealthCheckerTest, NoConnectionReuse` for testing in `thrift_test.cc`, which is the only test case for *not* reusing connection